### PR TITLE
SwiGLU optimized fw/bw

### PR DESCRIPTION
**Pull request by danthe3rd**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #518
* #511
* #510
* #509
* #505
* #508
* #504
* #491
* #502
* __->__ #490
* #498
* #488
* #487
* #486

**NOTE**
We can improve a bit more once this is fixed - https://github.com/NVIDIA/cutlass/issues/674

**USAGE**

```python
import xformers.ops as xops

# NOTE: Important to use `unbind` from xformers for the bw pass!
w1, w2 = xops.unbind(
    w1w2.view([2, w1w2.shape[0] // 2, w1w2.shape[1]]),
    dim=0,
)
b1, b2 = xops.unbind(b1b2.view([2, b1b2.shape[0] // 2]), dim=0)
y = xops.functional_swiglu(x,
    w1, b1, w2, b2, w3, b3)
```

**PERFORMANCE (A100 only)**

*FW*
```
[-------------------------------------------------------- swiglu_fw ---------------------------------------------------------]
                                     |  SwiGLUPackedFusedOp[fused.p.cpp]  |  eager   |  SwiGLUFusedOp[fused]
1 threads: -------------------------------------------------------------------------------------------------------------------
      f16    B=9456, I=1536, H=4096  |               1377.7               |  1581.4  |         1339.1
      f16.ac B=9456, I=1536, H=4096  |               1449.3               |  1735.3  |         1462.9
      f16    B=4440, I=1536, H=4096  |                600.4               |   735.6  |          593.9
      f16.ac B=4440, I=1536, H=4096  |                709.0               |   843.7  |          717.6
      f16    B=4728, I=1536, H=4096  |                638.9               |   776.2  |          635.3
      f16.ac B=4728, I=1536, H=4096  |                748.9               |   892.2  |          756.7
      f16    B=4728, I=1536, H=1024  |                162.3               |   201.5  |          163.1
      f16.ac B=4728, I=1536, H=1024  |                235.2               |   277.4  |          245.5

Times are in microseconds (us).
```

*BW*
```
[-------------------------------------------------------- swiglu_bw ---------------------------------------------------------]
                                     |  SwiGLUPackedFusedOp[fused.p.cpp]  |  eager   |  SwiGLUFusedOp[fused]
1 threads: -------------------------------------------------------------------------------------------------------------------
      f16    B=9456, I=1536, H=4096  |               2333.1               |  2696.7  |         2336.1
      f16.ac B=9456, I=1536, H=4096  |               2620.8               |  2990.9  |         2840.0
      f16    B=4440, I=1536, H=4096  |               1243.2               |  1413.8  |         1240.3
      f16.ac B=4440, I=1536, H=4096  |               1448.6               |  1629.0  |         1637.3
      f16    B=4728, I=1536, H=4096  |               1298.4               |  1481.5  |         1301.1
      f16.ac B=4728, I=1536, H=4096  |               1511.8               |  1705.3  |         1705.4
      f16    B=4728, I=1536, H=1024  |                463.3               |   493.9  |          463.0
      f16.ac B=4728, I=1536, H=1024  |                582.4               |   614.9  |          672.7

Times are in microseconds (us).
```

### CODE REVIEW

File: xformers/components/swiglu/cuda/silu_bw_fused.cu [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+template <>
+struct KernelTraits<at::Half> {
+  using AccumulationElement = float;
+};
+
+template <>
+struct KernelTraits<at::BFloat16> {
+  using AccumulationElement = float;
+};
```
Comment (fmassa) [inline xformers/components/swiglu/cuda/silu_bw_fused.cu:41]: nit: you can use `AccumulateType` (or `acc_type` for short) [from PyTorch here instead](https://github.com/pytorch/pytorch/blob/36380897553063c7b433f738671ff23c5ad58ced/aten/src/ATen/AccumulateType.h)

File: xformers/info.py [hunk 2]
```diff
@@ -20,6 +20,8 @@ def get_features_status() -> Dict[str, str]:
     features = {}
     for op in ALL_OPS:
         features[f"memory_efficient_attention.{op.NAME}"] = op.info()
+    for k, v in ops.swiglu._info().items():
+        features[f"swiglu.{k}"] = v
```
Comment (fmassa) [inline xformers/info.py:24]: nit: this is not scalable as we introduce more features

**Review by fmassa — APPROVED** [review]
Thanks!

File: xformers/components/attention/csrc/cuda/sddmm2_cuda.cu [hunk 3]
```diff
@@ -5,7 +5,7 @@
 #include <ATen/cuda/CUDAContext.h>
 #include <c10/cuda/CUDAGuard.h>
 #include <torch/types.h>
-#include "computeUtil.h"
+#include "../computeUtil.h"
```
[the remark this answers is not in the document]
Comment (danthe3rd) [reply xformers/components/attention/csrc/cuda/sddmm2_cuda.cu:8]: Nope but it's its current location, so the change looks correct. Now why did it use to build before, I don't know

**Pull request merged**